sprocs/sync_assessments.sql

```diff
@@ -84,12 +86,25 @@ BEGIN
             constant_question_value = EXCLUDED.constant_question_value,
             allow_issue_reporting = EXCLUDED.allow_issue_reporting,
             allow_real_time_grading = EXCLUDED.allow_real_time_grading,
-            require_honor_code = EXCLUDED.require_honor_code
+            require_honor_code = EXCLUDED.require_honor_code,
+            group_work = EXCLUDED.group_work    
         WHERE
             assessments.course_instance_id = sync_assessments.new_course_instance_id
         RETURNING id INTO new_assessment_id;
         new_assessment_ids = array_append(new_assessment_ids, new_assessment_id);
 
+        -- if it is a group work AND group_configs does not exist, insert new
+        PERFORM 1 FROM group_configs WHERE assessment_id = new_assessment_id AND deleted_at IS NULL;
```

Comment:
Maybe an obvious question, but what is this `PERFORM 1` line doing here?

Reply:
It is used to check whether there is a group_config in the database
If not found, insert a new group_config for that assessment

Reply:
Maybe it is the same as SELECT? Should I change it back?

Reply:
Inside sprocs you can't have SELECTS that don't store the results. Here we don't care about the results (only whether FOUND is true), so a PERFORM is reasonable.

However, this pattern of "check and then insert" is a standard anti-pattern, because it often results in race conditions. In this case, it looks like we are trying to ensure that we never have more than one non-deleted group config per assessment? If that's the case, it's normally better to add a DB uniqueness constraint on the table, and then use `INSERT ... ON CONFLICT DO NOTHING`.

Reply:
Yeah, it is reasonable, I will add a uniqueness constraint